hopsworks-web/yo/app/scripts/controllers/newStorageConnectorCtrl.js

```diff
@@ -230,19 +266,58 @@ angular.module('hopsWorksApp')
                         self.storageConnectorConfigWrongValue = -1
                     }
 
-                    if (self.s3AccessKey && self.s3AccessKey != null
-                        && self.s3AccessKey.length > self.s3StorageConnectorAccesskeyMaxLength) {
+                    if (self.s3ServerAndAccessKeys && (self.s3AccessKey === ""
+                        || self.s3AccessKey === null || self.s3AccessKey === undefined)) {
                         self.storageConnectorS3AccessKeyWrongValue = -1;
                         self.wrong_values = -1;
                         self.storageConnectorConfigWrongValue = -1
                     }
 
-                    if (self.s3SecretKey && self.s3SecretKey != null
-                        && self.s3SecretKey.length > self.s3StorageConnectorSecretkeyMaxLength) {
+                    if (self.s3ServerAndAccessKeys && self.s3AccessKey != null && self.s3AccessKey != undefined) {
+                        if(self.s3AccessKey.length > self.s3StorageConnectorAccesskeyMaxLength){
+                            self.storageConnectorS3AccessKeyWrongValue = -1;
+                            self.wrong_values = -1;
+                            self.storageConnectorConfigWrongValue = -1
+                        }
+                    }
```

Comment:
Conditions here  are repeated. Please simplify these ifs. Also you are not accounting for IAM roles.